src/components/CollectionProvider/CollectionProvider.tsx

```diff
@@ -0,0 +1,23 @@
+import * as React from 'react'
+import { Props } from './CollectionProvider.types'
+
+export default class CollectionProvider extends React.PureComponent<Props> {
+  componentDidMount() {
+    const { id, collection, onFetchCollection } = this.props
+    if (id && !collection) {
+      onFetchCollection(id)
+    }
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    const { id, onFetchCollection } = this.props
+    if (id && id !== prevProps.id) {
```

Comment:
should we check if the collection is in memory already before triggering the fetch here?

Reply:
I thought about it, but I'm not sure. I left it out because I maybe updating the local version could be useful? what do you think?

Reply:
good point, I guess it doesn't hurt to refresh it